docs/source/usage_guides/fsdp.md

```diff
@@ -161,6 +161,22 @@ When using transformers `save_pretrained`, pass `state_dict=accelerator.get_stat
 
 You can then pass `state` into the `save_pretrained` method.  There are several modes for `StateDictType` and `FullStateDictConfig` that you can use to control the behavior of `state_dict`.  For more information, see the [PyTorch documentation](https://pytorch.org/docs/stable/fsdp.html).
 
+If you choose to use `StateDictType.SHARDED_STATE_DICT`, the weights and optimizer states of the model during `Accelerator.save_state` will be split into `n` files for each sub-split on the model. To merge them back into
+a single dictionary to load back into the model later after training you can use the `merge_weights` utility:
+
+```py
+from accelerate.utils import merge_fsdp_weights
+
+# Our weights are saved usually in a `pytorch_model_fsdp_{model_number}` folder
+merge_fsdp_weights("pytorch_model_fsdp_0", "output_path", use_safetensors=True)
+```
+The final output will then either be saved to `merged.safetensors` or `merged.bin` (if `use_safetensors=False` is passed). 
```

Comment:
The issue is it's not just a model, it can be a model *or* an optimizer we're merging. :) (And no way to explicitly know)

Reply:
Thx for the explanation ! What would be the goal to merge the optimizer ? Retrain the model on another setup ? Also what happens if both the model and the optimizer are on the same folder. Maybe we can add something to the fsdp docs if it is useful. Otherwise, I would just focus on the model merging feature.